libs/vkd3d/state.c

```diff
@@ -1381,10 +1382,14 @@ static HRESULT d3d12_root_signature_init_global(struct d3d12_root_signature *roo
 
     root_signature->num_set_layouts = context.vk_set;
 
+    graphics_stages = VK_SHADER_STAGE_ALL_GRAPHICS;
+
+    if (device->device_info.mesh_shader_features.meshShader && device->device_info.mesh_shader_features.taskShader)
```

Comment:
I think we can use the same schism for compute / traceRay to separate legacy / mesh here. We'd just need to apply the root signature type in SetPipeline and potentially invalidate root parameters.